File: Codeforces/579_div3/C.cpp
#include<bits/stdc++.h>
#define ll long long 
using namespace std;
ll gcd(ll a,ll b){
	if(b==0){
		return a;
	}
	return gcd(b,a%b);
}
ll GCD(vector<ll> a){
	ll g = a[0];
	for(int i=1;i<a.size();i++){
		g=gcd(g,a[i]);
	}
	return g;
}
int main(){
	int n;
	cin>>n;
	vector<ll> a(n,0);
	for(int i=0;i<n;i++){
		cin>>a[i];
	}
	ll g = GCD(a);
	map<ll,int> m;
	while(g%2==0){
		g=g/2;
		m[2]++;
	}
	for(ll i=3;i*i<=g;i+=2){
		while(g%i==0){
			g=g/i;
			m[i]++;
		}
	}
	if(g>1){
		m[g]++;
	}
	ll pro=1;
	for(auto x:m){
		if(x.second!=0){
			pro=pro*(x.second+1);
		}
		
	}
	cout<<pro<<endl;
	return 0;
}

File: Codeforces/579_div3/C.py
def gcd(a,b):
	if(b==0):
		return a
	return gcd(b,a%b)
def GCD(a):
	gc=a[0]
	for i in range(1,len(a)):
		gc=gcd(gc,a[i])
	return gc
n=int(input())
a=list(map(int,input().split()))
G=GCD(a)
#print(G)
d=[0]*(G+1)
while(G%2==0):
	d[2]+=1
	G=G//2
G_copy=G
for i in range(3,G_copy+1,2):
	if(i>G):
		break
	while(G%i==0):
		d[i]+=1
		G=G//i
pro=1
for x in d:
	if(x!=0):
		pro = pro *(x+1)
print(pro)